cmd/segment-reaper/detect.go

```diff
@@ -212,14 +243,89 @@ func cmdDetect(cmd *cobra.Command, args []string) (err error) {
 	log.Info("number of inline segments", zap.Int("segments", observer.inlineSegments))
 	log.Info("number of last inline segments", zap.Int("segments", observer.lastInlineSegments))
 	log.Info("number of remote segments", zap.Int("segments", observer.remoteSegments))
+	log.Info("number of all segments", zap.Int("segments", observer.remoteSegments+observer.inlineSegments))
 	return nil
 }
 
 func analyzeProject(ctx context.Context, db metainfo.PointerDB, objectsMap ObjectsMap, csvWriter *csv.Writer) error {
-	// TODO this part will be implemented in next PR
+	for cluster, objects := range objectsMap {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			segments := make([]byte, 0)
+			for i := byte(0); i < maxNumOfSegments; i++ {
+				found := object.segments&(1<<uint64(i)) != 0
+				if found {
+					segments = append(segments, i)
+				}
+			}
```

Comment:
It looks like we are doing just to analyze later what bits are set in the `object.segments` bitmask. This would create garbage on every iteration (we would expect millions of iterations) and it would be difficult to implement unit tests for this logic later.

I would suggest creating a new type for the `object.segments` bitmask, based on `unit64`. We should encapsulate all bitmask operations in that type. This would make the code clearer, easier to implement unit tests, and easier to optimize the code for performance and memory garbage.